### PR TITLE
Completed Authentication Screens

### PR DESCRIPTION
**Pull request by aliarbaaz**
# Pull Request

## Description

- Implemented the Start screen, Login screen and Signup screen UI.
- Added navigation functionality between the Start, Login and Signup screens.

## Checklist

- [x] I have added a screenshot or screen recording showing the functionality of the app.
- [x] I have tested the project and confirmed it works as expected.

## Screenshots

start screen

login screen

login screen

### CODE REVIEW

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/LoginScreen.kt [hunk 1]
```diff
@@ -3,40 +3,149 @@ package com.android.engineer.mealmate.view.features.auth
...
+                            text = stringResource(id = R.string.login_to_account),
+                            modifier = Modifier,
+                            fontSize = 12.sp,
+                            textAlign = TextAlign.Center
+                        )
+                        Spacer(modifier = Modifier.height(16.dp))
+                        MealTextField(
+                            value = "",
+                            onValueChange = {},
+                            label = "Username",
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/LoginScreen.kt]: @aliarbaaz Get this string from strings.xml

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/SignupScreen.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                            value = "",
+                            onValueChange = {},
+                            label = "Username",
+                            icon = Icons.Filled.Person
+                        )
+                        Spacer(modifier = Modifier.height(16.dp))
+                        MealTextField(
+                            value = "",
+                            onValueChange = {},
+                            label = "First Name",
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/SignupScreen.kt]: Don't hard code the values.

**Review by Siduchels — CHANGES_REQUESTED** [review]

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/SignupScreen.kt (hunk 2 above)
Comment (aliarbaaz) [reply meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/SignupScreen.kt]: only label right?

Comment (Siduchels) [reply meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/auth/SignupScreen.kt]: Yes @aliarbaaz only for label value.

**Review by Siduchels — APPROVED** [review]

**Review by ZSank — APPROVED** [review]

**Pull request merged**